components/brave_ads/browser/ads_service_impl.h

```diff
@@ -91,6 +90,8 @@ class AdsServiceImpl : public AdsService,
 
   ~AdsServiceImpl() override;
 
+  base::WeakPtr<AdsServiceImpl> AsWeakPtr();
```

Comment:
Do we need this method to be public? We don't use weak pointers to AdsService outside of the class.

Reply:
It was public before. This PR is not making any functional changes to the code overall.

Reply:
I agree, thanks! AsWeakPtr() was in public prior to this change.
I actually was confused by BackgroundHelperLinux class where `AsWeakPtr()` was made private.

Reply:
Ah, yeah. I think in general it is inconsequential. I did try to not make changes, but I guess it slipped through.